kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
+                              '\'chmod 600 {}\''.format(self.config))
+
+                with open(self.config, 'r') as f:
+                    config_data = json.load(f)
+            except:
+                pass
+
+        else:
+
+            # Warn the user that configuration will be reliant on environment
+            if not quiet:
+                print('No Kaggle API config file found, will use environment.')
+
+        return config_data
+
+  
+    def _read_config(self):
+        '''read in the configuration file, a json file defined at self.confi'''
+        
         try:
             with open(self.config, 'r') as f:
                 config_data = json.load(f)
-            if value is not None:
-                config_data[name] = value
-                with open(self.config, 'w') as f:
-                    json.dump(config_data, f, indent=2)
-            if name in config_data:
-                self.config_values[name] = config_data[name]
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        except FileNotFoundError:
+            config_data = {}
+
+        return config_data
+
+
+    def _write_config(self, config_data, indent=1):
+        '''write config data to file.
+        '''
+        with open(self.config, 'w') as f:
+            json.dump(config_data, f, indent=indent)
+
+
+    def set_config_value(self, name, value, quiet=False):
+        '''a client helper function to set a configuration value, meaning
+           reading in the configuration file (if it exists), saving a new
+           config value, and then writing back
+        '''
+
+        config_data = self._read_config()
+   
+        # If defined by client, set and save!
+        self._write_config(config_data)
+
+        if value is not None:
+
+            # Update the config file with the value
+            config_data[name] = value
+
+            # Update the instance with the value
+            self.config_values[name] = value
+
+            if not quiet:
+                self.print_config_value(name, separator=' is now set to: ')
+
 
     def unset_config_value(self, name, quiet=False):
-        try:
-            with open(self.config, 'r') as f:
-                config_data = json.load(f)
-            config_data[name] = None
-            with open(self.config, 'w') as f:
-                json.dump(config_data, f, indent=2)
-            self.config_values[name] = None
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        '''unset a configuration value'''
 
-    def copy_config_value(self, name, values):
-        if name in values:
-            self.config_values[name] = values[name]
+        config_data = self._read_config()
+
+        # Remove it, if exists, both from loaded file and client
+
+        del self.config_values[name] # is there reason this was None?
```

Comment:
No idea, written by someone else before I joined the team who is now no longer with Kaggle.  If this works fine in your testing, then it's fine by me.